spec/datadog/core/workers/polling_spec.rb

```diff
@@ -22,25 +22,14 @@
       let(:task) { proc { |*args| worker_spy.perform(*args) } }
       let(:worker_spy) { double('worker spy') }
 
-      before { allow(worker_spy).to receive(:perform) { sleep 5 } }
-
-      context 'by default' do
-        before { skip('TODO: This test sometimes hangs forever and other times flakes. Requires further investigation.') }
-
-        it do
-          perform
-          try_wait_until { worker.running? && worker.run_loop? }
-          expect(worker_spy).to have_received(:perform).at_least(:once)
-        end
-      end
+      before { allow(worker_spy).to receive(:perform) }
 
       context 'when #enabled? is true' do
         before { allow(worker).to receive(:enabled?).and_return(true) }
 
         it do
           perform
-          try_wait_until { worker.running? && worker.run_loop? }
-          expect(worker_spy).to have_received(:perform).at_least(:once)
+          wait_for(worker_spy).to have_received(:perform)
```

Comment:
I like rspec-wait and I've used it in the past, but... it does seem to be a bit heavy to add a new dependency only for this single use-case.

Here's an alternative solution without 

```diff
diff --git a/spec/datadog/core/workers/polling_spec.rb b/spec/datadog/core/workers/polling_spec.rb
index 4b72daf1a..b4c839ac2 100644
--- a/spec/datadog/core/workers/polling_spec.rb
+++ b/spec/datadog/core/workers/polling_spec.rb
@@ -22,14 +22,17 @@ RSpec.describe Datadog::Core::Workers::Polling do
       let(:task) { proc { |*args| worker_spy.perform(*args) } }
       let(:worker_spy) { double('worker spy') }
 
-      before { allow(worker_spy).to receive(:perform) }
+      before do
+        @perform_called = false
+        allow(worker_spy).to receive(:perform) { @perform_called = true }
+      end
 
       context 'when #enabled? is true' do
         before { allow(worker).to receive(:enabled?).and_return(true) }
 
         it do
           perform
-          wait_for(worker_spy).to have_received(:perform)
+          try_wait_until { @perform_called }
         end
       end
```